Updating function to get cpuid and using latest snpguest release

DGonzalezVillal opened the pull request:
- Updated SNPGUEST to use latest version 0.7.1
- Updated necessary snpguest functions to be compatible with latest
  release
- Created a get_cpuid function to be able to read cpuid registers
- Used new cpuid function to create a new method to get host proc model

DGonzalezVillal committed:
Use CPUID to find the model name of host CPU

- Created a get_cpuid function to be able to read cpuid registers
- Used new cpuid function to create a new method to get host proc model

Signed-off-by: DGonzalezVillal <Diego.GonzalezVillalobos@amd.com>

ryansavino commented on tools/snp.sh:
```diff
@@ -95,7 +95,7 @@ AMDSEV_URL="https://github.com/ryansavino/AMDSEV.git"
 AMDSEV_NON_UPM_BRANCH="snp-non-upm"
 SNPGUEST_URL="https://github.com/virtee/snpguest.git"
-SNPGUEST_BRANCH="tags/v0.3.2"
+SNPGUEST_BRANCH="tags/v0.5.1"
```
Please put this in a separate commit so that the rest of this can be cherry picked into the coco branch.

ryansavino commented on tools/snp.sh:
```diff
@@ -268,6 +268,9 @@ install_dependencies() {
   # pip needed for sev-snp-measure
   sudo apt install -y python3-pip
+
+  # needed to find infomration about cpu
```
Comment grammar and spelling (suggest adding a spell checker to your IDE):
```
# Needed to find information about CPU
```

ryansavino commented on tools/snp.sh:
```diff
@@ -923,24 +926,101 @@ setup_guest_attestation() {
-  local cpu_model=$(cat /proc/cpuinfo | grep "model name" | head -1 | cut -d ' ' -f5)
-  local cpu_code_name="milan"
+# Pass a function and a register to collect its value
+get_cpuid() {
```
Variables in braces, keep spacing indentation consistent for this function and the other functions.

ryansavino commented on tools/snp.sh:
```diff
@@ -923,24 +926,101 @@ setup_guest_attestation() {
+            fi
+            ;;
+        *)
+            echo "invalid cpu"
```
Keep message returns in proper grammar:
```
echo "Invalid CPU"
```

ryansavino commented on tools/snp.sh:
```diff
@@ -1002,7 +1082,7 @@ attest_guest() {
 
   # Retrieve ark, ask, vcek (saved in ./certs)
-  ssh_guest_command "./snpguest fetch ca pem ${cpu_code_name} ."
+  ssh_guest_command "./snpguest fetch ca pem ${cpu_code_name} . --endorser vcek"
```
In separate commit per first review comment.

ryansavino reviewed (changes requested):
Few nits.

DGonzalezVillal committed:
Updating snpguest to latest release

- Updated SNPGUEST to use latest version 0.6.0
- Updated necessary snpguest functions to be compatible with latest
release

Signed-off-by: DGonzalezVillal <Diego.GonzalezVillalobos@amd.com>

larrydewey commented on tools/snp.sh:
```diff
@@ -95,7 +95,7 @@ AMDSEV_URL="https://github.com/ryansavino/AMDSEV.git"
 AMDSEV_NON_UPM_BRANCH="snp-non-upm"
 SNPGUEST_URL="https://github.com/virtee/snpguest.git"
-SNPGUEST_BRANCH="tags/v0.3.2"
+SNPGUEST_BRANCH="tags/v0.5.2"
```
```suggestion
SNPGUEST_BRANCH="tags/v0.6.0"
```
